zap/src/main/java/org/zaproxy/zap/extension/ascan/FilterPanel.java

```diff
@@ -0,0 +1,496 @@
+            if (getTagModel().contains(sel)) {
+                inds[i] = getTagModel().indexOf(sel);
+            }
+            i++;
+        }
+        incTagList.setSelectedIndices(inds);
+    }
+
+    private void addTagsInFilterPanel(SiteNode siteNode, Collection<String> tags) {
+        if (siteNode != null) {
+            if (siteNode.getHistoryReference() != null
+                    && siteNode.getHistoryReference().getTags() != null) {
+                tags.addAll(siteNode.getHistoryReference().getTags());
+            }
+            Enumeration<?> childEnumeration = siteNode.children();
+            while (childEnumeration.hasMoreElements()) {
+                Object node = childEnumeration.nextElement();
+                if (node instanceof SiteNode) {
+                    addTagsInFilterPanel((SiteNode) node, tags);
+                }
+            }
+        }
+    }
+
+    private void populateTagsInFilterPanel(Target target) {
+        Set<String> tags = new LinkedHashSet<>();
+        if (target != null) {
+            SiteNode siteNode = target.getStartNode();
+            if (siteNode != null) {
+                this.addTagsInFilterPanel(siteNode, tags);
+            } else if (target.getContext() != null || target.isInScopeOnly()) {
+                List<SiteNode> nodes = Collections.emptyList();
+                if (target.isInScopeOnly()) {
+                    nodes = Model.getSingleton().getSession().getTopNodesInScopeFromSiteTree();
+                } else if (target.getContext() != null) {
+                    nodes = target.getContext().getTopNodesInContextFromSiteTree();
+                }
+                for (SiteNode node : nodes) {
+                    this.addTagsInFilterPanel(node, tags);
+                }
+            }
+        }
+        this.setAllTags(new ArrayList<String>(tags));
+    }
+
+    public List<ScanFilter> getScanFilters() {
+        List<ScanFilter> scanFilterList = new ArrayList<>();
+        try {
+            scanFilterList.addAll(
+                    GenericFilterUtility.getScanFilter(
+                            incTagList.getSelectedValuesList(),
+                            excTagList.getSelectedValuesList(),
+                            TagScanFilter::new));
+            scanFilterList.addAll(
+                    GenericFilterUtility.getScanFilter(
+                            methodList.getSelectedValuesList(), null, MethodScanFilter::new));
+            scanFilterList.addAll(
+                    GenericFilterUtility.getScanFilter(
+                            codeList.getSelectedValuesList(), null, HttpStatusCodeScanFilter::new));
+            scanFilterList.addAll(
+                    GenericFilterUtility.getScanFilter(
+                            strToRegexList(regexInc.getText()),
+                            strToRegexList(regexExc.getText()),
+                            UrlPatternScanFilter::new));
+        } catch (PatternSyntaxException e1) {
```

Comment:
Should not catch to just re-throw.

Reply:
oh .. i think i missed to remove this check. will do the change